### PR TITLE
EPMRPP-36944 - React. Admin part. All users. Control with projects

### PR DESCRIPTION
**Pull request by mykhailo-kalashnikov-epam**

### CODE REVIEW

File: app/src/pages/admin/allUsersPage/allUsersGrid/projectsAndRolesColumn/projectsAndRolesColumn.jsx [hunk 1]
```diff
@@ -1,31 +1,261 @@
...
+    this.setState({
+      isMobileView: media.matches,
+    });
+  getProjectsList = () => {
+    const {
+      value: { assignedProjects = {} },
+    } = this.props;
+    return Object.keys(assignedProjects);
+  };
+  getVisibleProjects = () => this.getProjectsList().slice(0, 2);
```
Comment (margalencia) [inline app/src/pages/admin/allUsersPage/allUsersGrid/projectsAndRolesColumn/projectsAndRolesColumn.jsx]: why there are only two first projects are visible?

Comment (mykhailo-kalashnikov-epam) [reply app/src/pages/admin/allUsersPage/allUsersGrid/projectsAndRolesColumn/projectsAndRolesColumn.jsx]: That's business requirements

**Comment by mykhailo-kalashnikov-epam** [comment]
> It looks like you forgot to add translations to be.json end ru.json

Done.

File: app/src/pages/admin/allUsersPage/allUsersGrid/projectsAndRolesColumn/projectsAndRolesColumn.jsx [hunk 2]
```diff
@@ -1,31 +1,261 @@
...
+  unAssignTitle: {
+    id: 'projectsAndRolesColumn.unAssignTitle',
+    defaultMessage: 'Unassign user from the project',
+  },
+  addProject: {
+    id: 'projectsAndRolesColumn.addProject',
+    defaultMessage: '+ Add Project',
+  },
+});
+@connect(() => ({}), { showNotification, showModalAction, fetchAllUsersAction, toggleUserRoleForm })
```
Comment (AmsterGet) [inline app/src/pages/admin/allUsersPage/allUsersGrid/projectsAndRolesColumn/projectsAndRolesColumn.jsx]: `() => ({})` can be just `null`

File: app/src/pages/admin/allUsersPage/allUsersGrid/projectsAndRolesColumn/projectsAndRolesColumn.scss [hunk 3]
```diff
@@ -1,88 +1,136 @@
...
-}
-.project-role {
-  text-transform: capitalize;
-}
+.projects-and-roles-col {
+  width: 25%;
+  vertical-align: middle;
+  font-size: 13px;
+  line-height: 20px;
+  // overflow: hidden;
```
Comment (AmsterGet) [inline app/src/pages/admin/allUsersPage/allUsersGrid/projectsAndRolesColumn/projectsAndRolesColumn.scss]: Please remove if unnecessary

File: app/src/controllers/administrate/allUsers/actionCreators.js [hunk 4]
```diff
@@ -16,6 +16,14 @@ export const toggleUserSelectionAction = toggleItemSelectionAction(NAMESPACE);
 export const toggleAllUsersAction = toggleAllItemsAction(NAMESPACE);
 export const unselectAllUsersAction = unselectAllItemsAction(NAMESPACE);
 
+export const toggleUserRoleForm = (userId, value) => ({
```
Comment (AmsterGet) [inline app/src/controllers/administrate/allUsers/actionCreators.js]: `toggleUserRoleFormAction`

File: app/src/controllers/administrate/allUsers/reducer.js [hunk 5]
```diff
@@ -3,10 +3,25 @@ import { fetchReducer } from 'controllers/fetch';
 import { paginationReducer } from 'controllers/pagination';
 import { loadingReducer } from 'controllers/loading';
 import { groupOperationsReducer } from 'controllers/groupOperations';
-import { NAMESPACE } from './constants';
+import { queueReducers } from 'common/utils/queueReducers';
+import { NAMESPACE, TOGGLE_USER_ROLE_FORM } from './constants';
+
+const toggleUserRoleForm = (state, { type, payload = {} }) => {
```
Comment (AmsterGet) [inline app/src/controllers/administrate/allUsers/reducer.js]: `toggleUserRoleFormReducer`

**Pull request merged**